include/mitsuba/core/distr_1d.h

```diff
@@ -392,32 +414,40 @@ template <typename Value> struct ContinuousDistribution {
      * \brief %Transform a uniformly distributed sample to the stored
      * distribution
      *
-     * \param value
+     * \param sample
      *     A uniformly distributed sample on the interval [0, 1].
      *
      * \return
      *     The sampled position.
      */
-    Value sample(Value value, Mask active = true) const {
+    Value sample(Value sample, Mask active = true) const {
         MI_MASK_ARGUMENT(active);
 
-        value *= m_integral;
+        sample *= m_integral;
 
         Index index = dr::binary_search<Index>(
             m_valid.x(), m_valid.y(),
             [&](Index index) DRJIT_INLINE_LAMBDA {
-                return dr::gather<Value>(m_cdf, index, active) < value;
+                //return dr::gather<Value>(m_cdf, index, active) < value;
+                Value value = dr::gather<Value>(m_cdf, index, active);
+                if constexpr (!dr::is_jit_v<Float>) {
+                    return value < sample;
+                } else {
+                    // `m_valid` did not trim the leading/traling zeros in the PMF
+                    return ((value < sample) || (dr::eq(value, 0))) &&
+                           dr::neq(value, m_integral);
```

Comment:
Here, there is a check against `m_integral`. Above, it was against `m_normalization`. Is that right?